packages/routing-plugin/src/index.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  createUniverseWatcher,
+  PluginActionCtx,
+  PluginDispatch,
+  PluginViewCtx,
+  ProdoPlugin,
+} from "@prodo/core";
+import { createBrowserHistory, createMemoryHistory, History } from "history";
+
+export interface RouteParams {
+  path: string;
+  params?: { [key: string]: string };
+}
+
+const historySymbol = Symbol("@@routing/history");
+const universeSymbol = Symbol("@@routing/universe");
+
+export interface Universe {
+  route: RouteParams;
+}
+
+export interface Routing {
+  [historySymbol]: History;
+  [universeSymbol]: Universe;
+  routing: {
+    push: (params: RouteParams) => void;
+    replace: (params: RouteParams) => void;
+  };
+  route: RouteParams;
+}
+
+export interface Config {
+  routing?: {
+    testMode?: boolean;
```

Comment:
How do we feel about this? We could also use a global `testMode` flag (to avoid plugins having to manage this themselves) or we could instead force the user to pass a `history` themselves (this is more similar to how other frameworks do it, since you can then create other more optimised histories etc.

(e.g. in react-router you do

```tsx
const history = createBrowserHistory();
render(<Router history={history}><App /></Router>);
```

and in `connected-react-router` you do

```tsx
const history = createBrowserHistory();
const store = createStore(
  createReducer(history),
  applyMiddleware(routingMiddleware(history)),
);
render(<ConnectedRouter history={history}><App /></ConnectedRouter>);
```

Reply:
I personally prefer the `testMode` flag instead of passing a separate history, but don't mind either way.

To prevent every plugin from having a separate `testMode` config option, we could move that up to the top level so plugins could "share" config. If the convention is for plugins to use `config.testMode` then the user will only have to supply it once.

Reply:
I decided to pass a `history` because it's closer to what other libraries do, but I'd also be happy to manage this ourselves, via a root-level `testMode` flag.